jacodb-analysis/src/main/kotlin/org/jacodb/analysis/codegen/language/impl/JavaLanguage.kt

```diff
@@ -390,18 +400,20 @@ class JavaLanguage : TargetLanguage {
         }
         appendSite(callable.preparationSite)
         val sites = callable.callSites
+        val hasDeref = !(callable.terminationSite.expressionsBefore.isEmpty()
+                && callable.terminationSite.expressionsAfter.isEmpty() && callable.terminationSite.dereferences.isEmpty())
```

Comment:
for all intermediate calls, including field access, separate variable should be created. 

This simplifies debugging and code reading